--- raid10.c	2020-02-25 04:21:26.778580704 +0000
+++ raid10.c	2020-02-25 04:21:01.090427812 +0000
@@ -4486,13 +4486,14 @@
 	/* Use sync reads to get the blocks from somewhere else */
 	int sectors = r10_bio->sectors;
 	struct r10conf *conf = mddev->private;
-	struct {
-		struct r10bio r10_bio;
-		struct r10dev devs[conf->copies];
-	} on_stack;
-	struct r10bio *r10b = &on_stack.r10_bio;
+	struct r10bio *r10b;
 	int slot = 0;
 	int idx = 0;
+ 	r10b = kmalloc(sizeof(*r10b) +
+ 	       sizeof(struct r10dev) * conf->copies, GFP_KERNEL);
+ 	if (!r10b)
+ 		return -ENOMEM;
+
 	struct bio_vec *bvec = r10_bio->master_bio->bi_io_vec;
 
 	r10b->sector = r10_bio->sector;
 
